src/Boot/BootstrapHook.php

```diff
@@ -24,7 +24,8 @@ public function initialize(InputInterface $input, AnnotationData $annotationData
     {
         // Get the @bootstrap annotation. If there isn't one, then assume NONE.
         $phase = $annotationData->get('bootstrap', 'none');
-        $bootstrap_successful = $this->bootstrapManager->bootstrapToPhase($phase);
+        $kernel = $annotationData->get('kernel', 'drupal');
+        $bootstrap_successful = $this->bootstrapManager->bootstrapToPhase($phase, $kernel);
```

Comment:
Bootstrap manager should not receive Drupal-specific data. Pass in the annotation data here, and let the Drupal 8 bootstrap object look up the Kernel from the annotations.

Reply:
Done.